src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -107,13 +108,24 @@ export async function suggest(
 
   const { getFieldsByType, getFieldsMap } = getFieldsByTypeRetriever(
     queryForFields.replace(EDITOR_MARKER, ''),
-    resourceRetriever
+    resourceRetriever,
+    innerText
   );
   const supportsControls = resourceRetriever?.canSuggestVariables?.() ?? false;
   const getVariables = resourceRetriever?.getVariables;
   const getSources = getSourcesHelper(resourceRetriever);
   const { getPolicies, getPolicyMetadata } = getPolicyRetriever(resourceRetriever);
 
+  const lastCharacterTyped = innerText[innerText.length - 1];
+  let controlSuggestions: SuggestionRawDefinition[] = [];
+  if (lastCharacterTyped === '?') {
+    controlSuggestions = getControlSuggestionIfSupported(
+      Boolean(supportsControls),
+      ESQLVariableType.VALUES,
+      getVariables
+    );
+  }
+
```

Comment:
Are you ever going to have controls outside of an expression? If not, it seems like this logic belongs in `suggestForExpression`, though we may have to wait until https://github.com/elastic/kibana/issues/216492 so that it can be fully supported. If you agree, can we add a `TODO` comment here?

Also, why not return the suggestion right here instead of letting the other routines complete?

Reply:
I am not sure how the feature will evolve.

> Also, why not return the suggestion right here instead of letting the other routines complete?

It is on purpose! I wanted mostly to avoid this

<img width="535" alt="image" src="https://github.com/user-attachments/assets/90a894e5-7ef0-4b33-9662-675dd646ffcc" />

Reply:
Makes sense, done https://github.com/elastic/kibana/pull/216839/commits/9bf9f345f253de84736f8ae36a37489432a3fe7e